src/main/java/ui/UI.java

```diff
@@ -540,4 +541,22 @@ public boolean isWindowMinimized() {
     public boolean isWindowFocused() {
         return mainStage.isFocused();
     }
+
+    private void checkJavaDependency() {
+        JavaVersion javaRuntimeVersion = JavaVersion.fromString(System.getProperty("java.runtime.version"));
+        JavaVersion javaRequiredVersion = JavaVersion.fromString(JavaVersion.REQUIRED_VERSION);
+
+        checkJavaDependency(javaRuntimeVersion, javaRequiredVersion);
+    }
+
+    private void checkJavaDependency(JavaVersion javaRuntimeVersion, JavaVersion javaRequiredVersion) {
+        if (javaRuntimeVersion.compareTo(javaRequiredVersion) < 0) {
```

Comment:
`If` blocks should capture odd cases. The normal case should not be nested. Reverse the condition and use an early return.